python/ray/scripts/scripts.py

```diff
@@ -964,122 +964,163 @@ def start(
     """Stop Ray processes manually on the local machine."""
-
-    # Note that raylet needs to exit before object store, otherwise
-    # it cannot exit gracefully.
     is_linux = sys.platform.startswith("linux")
-    processes_to_kill = RAY_PROCESSES
-
-    process_infos = []
-    for proc in psutil.process_iter(["name", "cmdline"]):
-        try:
-            process_infos.append((proc, proc.name(), proc.cmdline()))
-        except psutil.Error:
-            pass
-
-    stopped = []
-    for keyword, filter_by_cmd in processes_to_kill:
-        if filter_by_cmd and is_linux and len(keyword) > 15:
-            # getting here is an internal bug, so we do not use cli_logger
-            msg = (
-                "The filter string should not be more than {} "
-                "characters. Actual length: {}. Filter: {}"
-            ).format(15, len(keyword), keyword)
-            raise ValueError(msg)
-
-        found = []
-        for candidate in process_infos:
-            proc, proc_cmd, proc_args = candidate
-            corpus = proc_cmd if filter_by_cmd else subprocess.list2cmdline(proc_args)
-            if keyword in corpus:
-                found.append(candidate)
-        for proc, proc_cmd, proc_args in found:
-            proc_string = str(subprocess.list2cmdline(proc_args))
+    total_procs_found = 0
+    total_procs_stopped = 0
+    procs_not_gracefully_killed = []
+
+    def kill_procs(
+        force: bool, grace_period: int, processes_to_kill: List[str]
+    ) -> Tuple[int, int, List[psutil.Process]]:
+        """Find all processes from `processes_to_kill` and terminate them.
+
+        Unless `force` is specified, it gracefully kills processes. If
+        processes are not cleaned within `grace_period`, it force kill all
+        remaining processes.
+
+        Returns:
+            total_procs_found: Total number of processes found from
+                `processes_to_kill` is added.
+            total_procs_stopped: Total number of processes gracefully
+                stopped from `processes_to_kill` is added.
+            procs_not_gracefully_killed: If processes are not killed
+                gracefully, they are added here.
+        """
+        process_infos = []
+        for proc in psutil.process_iter(["name", "cmdline"]):
             try:
-                if force:
-                    proc.kill()
-                else:
-                    # TODO(mehrdadn): On Windows, this is forceful termination.
-                    # We don't want CTRL_BREAK_EVENT, because that would
-                    # terminate the entire process group. What to do?
-                    proc.terminate()
-
-                if force:
+                process_infos.append((proc, proc.name(), proc.cmdline()))
+            except psutil.Error:
+                pass
+
+        stopped = []
+        for keyword, filter_by_cmd in processes_to_kill:
+            if filter_by_cmd and is_linux and len(keyword) > 15:
```

Comment:
do u know why 15?

Reply:
not sure why... this was the code that was there before